Refactor/style changes

jerry1ye10 opened the pull request:
Made Style changes to align with indenting and 80 character limit per line

jerry1ye10 committed:
Update models.py

NicholasLYang committed:
took longer than expected, fixed some other issues

NicholasLYang committed:
oops

NicholasLYang committed:
fixing up some issues

NicholasLYang committed:
finishing touches

jerry1ye10 committed:
Update views.py

squirrelsquirrel78 commented on app/views.py:
```diff
@@ -3,129 +3,130 @@
 
 
-#----------------------------------------------    
+#----------------------------------------------  
```
please add a title to this section divider (for example, "Error Handlers").

jerry1ye10 committed:
Update views.py

squirrelsquirrel78 commented on app/views.py:
```diff
@@ -1,131 +1,134 @@
-
-@app.route('/newspaper/<int:volume>/<int:issue>', methods=['GET'])
+@app.route('/newspaper/<int:volume>/<int:issue>' )
 def retrieve_article_data(volume,issue):
```
this function name is ambiguous. specify in the function name how this article is being gotten.

squirrelsquirrel78 replied:
should be something along the lines of `get_issue_articles()`

squirrelsquirrel78 commented on app/views.py:
```diff
@@ -1,131 +1,134 @@
-    issuu_code = models.Issuu.query.filter(models.Issuu.volume == volume 
-        and models.Issuu.issue == issue).first().code
-    return jsonify({"issuu_code": issuu_code, "articles": secure_articles})
+  articles = models.Article.query.filter(models.Article.volume == volume 
```
i would name this something like articlesInIssue

squirrelsquirrel78 commented on app/views.py:
```diff
@@ -1,131 +1,134 @@
+    "content": article.content, 
+    "datetime": article.datetime, 
+    "id": article.id, 
+    "isDraft": article.isDraft, 
```
you changed this to snakecase in the model. should be "is_draft". this code is obviously not tested.

squirrelsquirrel78 commented on app/views.py:
```diff
@@ -1,131 +1,134 @@
 
-@app.route('/list_articles/articles/', methods=['GET'])
+@app.route('/list_articles/articles/' )
 def all_articles():
```
again with the CRUD verbs: this function should be `get_all_articles()`

squirrelsquirrel78 reviewed (commented):
Strongly recommend having a parent_slug in the Subsection table. Not only does it now mirror the structure of CFA, several routing exceptions can be caught. For example, in views.py:27-30, the request would work for a subsection even if the link were /sections/section_slug_that_does_not_have_subsection_as_child/subsections/subsection_slug

squirrelsquirrel78 commented on app/views.py:
```diff
@@ -1,131 +1,134 @@
 
-@app.route('/user/<int:user_id>', methods=['GET'])
+@app.route('/user/<int:user_id>')
 def retrieve_user_data(user_id):
```
should be `get_user()`. CRUD VERBSS

jerry1ye10 committed:
Update views.py

squirrelsquirrel78 commented on app/views.py:
```diff
@@ -1,131 +1,132 @@
-    return jsonify({"issuu_code": issuu_code, "articles": secure_articles})
+  articles = models.Article.query.filter(models.Article.volume == volume 
+    and models.Article.issue == issue).all()
+  secure_articles = []
```
why is this called secure_articles?

jerry1ye10 committed:
Update views.py

Fixed views.py based on Jason's comments

jerry1ye10 committed:
Update views.py

jerry1ye10 committed:
Update models.py

Changed Subsections to have a parent_slug column

jerry1ye10 committed:
Update db_seed.py

jerry1ye10 committed:
Update views.py

Updated to match alongside the updated subsection model with parent_slug as a attribute

jerry1ye10 committed:
Update db_seed.py
